scheduled/cru-org-feed.js

```diff
@@ -0,0 +1,162 @@
+        },
+        json: true
+      })
+    }
+
+    /**
+     * Query list of scores as a Stream
+     * @returns {Promise<void>}
+     */
+    const getScoreStream = async () => {
+      const postgresClient = new Client({
+        user: process.env.DB_ENV_POSTGRESQL_USER,
+        password: process.env.DB_ENV_POSTGRESQL_PASS,
+        database: process.env.DB_ENV_POSTGRESQL_DB,
+        host: process.env.DB_PORT_5432_TCP_ADDR || 'localhost',
+        port: process.env.DB_PORT_5432_TCP_PORT || 5432
+      })
+      await postgresClient.connect()
+      const query = new QueryStream(scoresQuery, [prefixMatch.source])
+      const scoreStream = postgresClient.query(query)
+      scoreStream.on('end', async () => {
+        await postgresClient.end()
+      })
+      return scoreStream
+    }
+
+    /**
+     * Adds a header to a stream
+     * @param {String} header
+     * @returns {module:stream.internal.Transform}
+     */
+    const addStreamHeader = (header) => {
+      let prepended = false
+      return new Transform({
+        objectMode: true,
+        transform: function (data, encoding, callback) {
+          if (!prepended) {
+            this.push(Buffer.from(header, 'utf8'))
+            prepended = true
+          }
+          callback(null, data)
+        }
+      })
+    }
+
+    const cruOrgJson = await getCruOrgJson()
+
+    const scoreTransform = CSV.transform((record, callback) => {
+      const path = record['entity.pageUrl'].replace(prefixMatch, '/content/cru').replace('.html', '')
+      if (path.indexOf('.') > -1) {
+        callback(null, null)
+      } else {
+        let page = find(cruOrgJson.hits, {'jcr:path': path})
+        if (page === undefined) {
+          callback(null, null)
+        } else {
+          let [language, primarycat, ...subcats] = path.split('/').slice(4, -1)
+          if (typeof page['jcr:content'] !== 'undefined') {
+            let content = page['jcr:content']
+            if (typeof content['jcr:title'] !== 'undefined') {
+              record['entity.name'] = content['jcr:title'].trim()
+            }
+            if (typeof content['jcr:description'] !== 'undefined') {
+              record['entity.message'] = content['jcr:description'].replace(new RegExp('[\\r\\n]', 'gi'), ' ').trim()
+            } else if (typeof content['subtitle'] !== 'undefined') {
+              record['entity.message'] = content['subtitle'].replace(new RegExp('[\\r\\n]', 'gi'), ' ').trim()
+            }
+            if (typeof content['image'] !== 'undefined' && typeof content['image']['fileReference'] !== 'undefined') {
+              record['entity.thumbnailUrl'] = record['entity.pageUrl'].replace(/\.html$/i, '/_jcr_content/image.transform/CruHalf432x243/img.png')
+            }
+          }
+          if (typeof primarycat !== 'undefined') {
+            record['entity.primary_category'] = primarycat
+          }
+          // drop last subcat as it's the article name
```

Comment:
Is this guaranteed by something, or is it just the case now?

Reply:
subcats is just url split by `/`, so last part is always the page/article name.